spec.emu

```diff
@@ -168,3 +168,334 @@ contributors: Surma, Dan Ehrenberg
+    <emu-table id="table-realm-record-fields" caption="Realm Record Fields" oldids="table-21">
+      <table>
+        <tr>
+          <th>
+            Field Name
+          </th>
+          <th>
+            Value
+          </th>
+          <th>
+            Meaning
+          </th>
+        </tr>
+        <tr>
+          <td>
+            [[Intrinsics]]
+          </td>
+          <td></td>
+          <td></td>
+        </tr>
+        <tr>
+          <td>
+            [[GlobalObject]]
+          </td>
+          <td></td>
+          <td></td>
+        </tr>
+        <tr>
+          <td>
+            [[GlobalEnv]]
+          </td>
+          <td></td>
+          <td></td>
+        </tr>
+        <tr>
+          <td>
+            [[TemplateMap]]
+          </td>
+          <td></td>
+          <td></td>
+        </tr>
+        <tr>
+          <td>
+            <ins>[[ModuleMap]]</ins>
+          </td>
+          <td>
+            <ins>a List of Record { [[ReferencingScriptOrModule]]: Script Record or Module Record or *null*, [[Specifier]]: String or Object, [[ModuleRecord]]: Module Record }</ins>
+          </td>
+          <td>
+            <p><ins>Module Records are cached separately per Realm: two imports from the same realm using the same specifier from the same referencing script or module must resolve to the same Module Record. The cache needs to be per-Realm to support cases when [[ReferencingScriptOrModule]] is *null*.</ins></p>
+            <p><ins>This List respects the following invariants:</ins></p>
+            <ul>
+              <li>
+                <ins>It does not contain two Records _record1_ and _record2_ such that _record1_.[[ReferencingScriptOrModule]] is _record2_.[[ReferencingScriptOrModule]] and _record1_.[[Specifier]] is _record2_.[[Specifier]].</ins>
+              </li>
+              <li>
+                <p><ins>For every Record _record_ it contains, if Type(_record_.[[Specifier]]) is Object then:</ins></p>
+                <ul>
+                  <li>
+                    <ins>_record_.[[Specifier]] has a [[ModuleBlockBody]] internal slot.</ins>
+                  </li>
+                  <li>
+                    <ins>_record_.[[ReferencingScriptOrModule]] is *null*.</ins>
+                  </li>
+                </ul>
+              </li>
+            </ul>
+          </td>
+        </tr>
+        <tr>
+          <td>
+            [[HostDefined]]
+          </td>
+          <td></td>
+          <td></td>
+        </tr>
+      </table>
+    </emu-table>
+  </emu-clause>
+</emu-clause>
+
+<emu-clause id="sec-ecmascript-language-scripts-and-modules">
+  <h1>ECMAScript Language: Scripts and Modules</h1>
+
+  <emu-clause id="sec-modules">
+    <h1>Modules</h1>
+
+    <emu-clause id="sec-module-semantics">
+      <h1>Module Semantics</h1>
+
+
+      <emu-clause id="sec-source-text-module-records">
+        <h1>Source Text Module Records</h1>
+
+        <emu-clause id="sec-parsemodule" type="abstract operation">
```

Comment:
Is the expectation that this will be used by other host functions as a refactoring? If so, is it worth doing that refactoring in this spec or is the plan to do it later on?

Reply:
ParseModule is already an existing abstract operation, I only modified it to accept a Parse Node (since module blocks are already "parsed") so that I can use to initialize module blocks.

Reply:
Oh I see what you meant. I didn't think about that, but I feel like it should be done in parallel regardless of this proposal.

Reply:
I actually misread this initially, sorry for the confusion!